internal/charmstore/server.go

```diff
@@ -89,44 +90,25 @@ type ServerParams struct {
 
 	// MinUploadPartSize holds the minimum size of
 	// an upload part. If it's zero, a default value will be used.
-	MinUploadPartSize int64 `json:"min-upload-part-size"`
+	MinUploadPartSize int64
 
 	// MaxUploadPartSize holds the maximum size of
 	// an upload part. If it's zero, a default value will be used.
-	MaxUploadPartSize int64 `json:"max-upload-part-size"`
+	MaxUploadPartSize int64
 
 	// MaxUploadParts holds the maximum number of upload
 	// parts that can be uploaded in a single upload.
 	// If it's zero, a default value will be used.
-	MaxUploadParts int `json:"max-upload-parts"`
+	MaxUploadParts int
 
 	// RunBlobStoreGC holds whether the server will run
 	// the blobstore garbage collector worker.
 	RunBlobStoreGC bool
 
-	// BlobStore holds the type of blobstore to use.
-	BlobStore string `yaml:"blobstore"`
-
-	// SwiftAuthURL holds the swift auth url, equivalent to OS_AUTH_URL.
-	SwiftAuthURL string `yaml:"swift-auth-url"`
-
-	// SwiftUsername holds the swift username, equivalent to OS_USERNAME..
-	SwiftUsername string `yaml:"swift-username"`
-
-	// SwiftSecret holds the swift password, equivalent to OS_PASSWORD.
-	SwiftSecret string `yaml:"swift-secret"`
-
-	// SwiftBucket holds the swift bucket to use for blobs.
-	SwiftBucket string `yaml:"swift-bucket"`
-
-	// SwiftRegion holds the swift region to use for blobs.
-	SwiftRegion string `yaml:"swift-region"`
-
-	// SwiftTenant holds the swift tenant to use for connecting to swift.
-	SwiftTenant string `yaml:"swift-tenant"`
-
-	// SwiftAuthMode holds the swift auth mode to use for connecting to swift.
-	SwiftAuthMode string `yaml:"swift-authmode"`
+	// NewBlobBackend returns a new blobstore backend
+	// that may use the given MongoDB database.
+	// If this is nil, a MongoDB backend will be used.
+	NewBlobBackend func(db *mgo.Database) blobstore.Backend
```

Comment:
I understand why this is here, but given the code you actually the value won't currently be used with any non-nil value for NewBlobBackend. It seems a real shame to leek the abstraction like that.